lib/config/config.go

```diff
@@ -56,6 +78,13 @@ func IsPerennialBranch(branchName string) bool {
 }
 
 
+func KnowsAllAncestorBranches(branchName string) bool {
+  return branchName == GetMainBranch() ||
+    IsPerennialBranch(branchName) ||
+    len(GetAncestorBranches(branchName)) > 0
```

Comment:
All we seem to care about here is whether there is a valid entry for ancestor branches in the Git config. Maybe create a helper called "HasStoredAncestorBranches" to make this code more expressive?